soc/xtensa/intel_s1000/soc.h

```diff
@@ -28,6 +28,17 @@
 #define INTR_CNTL_IRQ_NUM(_irq) \
 	(((_irq >> INTR_CNTL_IRQ_NUM_SHIFT) & INTR_CNTL_IRQ_NUM_MASK) - 1)
 
+/* Macro that aggregates the tri-level interrupt into an IRQ number */
+#define SOC_AGGREGATE_IRQ(ictl_irq, cavs_irq, core_irq)		\
+	(((core_irq & XTENSA_IRQ_NUM_MASK) << XTENSA_IRQ_NUM_SHIFT) |	\
+	(((cavs_irq) & CAVS_IRQ_NUM_MASK) << CAVS_IRQ_NUM_SHIFT) |	\
+	(((ictl_irq) & INTR_CNTL_IRQ_NUM_MASK) << INTR_CNTL_IRQ_NUM_SHIFT))
+
+#define CAVS_L2_AGG_INT_LEVEL2			DT_CAVS_ICTL_0_IRQ
+#define CAVS_L2_AGG_INT_LEVEL3			DT_CAVS_ICTL_1_IRQ
+#define CAVS_L2_AGG_INT_LEVEL4			DT_CAVS_ICTL_2_IRQ
+#define CAVS_L2_AGG_INT_LEVEL5			DT_CAVS_ICTL_3_IRQ
+
```

Comment:
It would be good if the changes to this file until the above point has its own commit. Then, in the subsequent commits, each of the interrupts should be defined in terms of  SOC_AGGREGATE_IRQ (like the way I2S interrupt is handled below).

Reply:
> It would be good if the changes to this file until the above point has its own commit.

I can do that.



> each of the interrupts should be defined in terms of SOC_AGGREGATE_IRQ (like the way I2S interrupt is handled below

I don't follow this comment. Are you asking to use the macro for defining the IRQ for other blocks as well -- the ones that you previously hard coded?

All of the IRQ definitions should go away from here and be moved to DTS, like what @galak has been doing, for example, in #12847 except that it should be done for all 3 values and them assembled using the macro.
It's for another PR.

Reply:
> > It would be good if the changes to this file until the above point has its own commit.
> 
> I can do that.
> 
> > each of the interrupts should be defined in terms of SOC_AGGREGATE_IRQ (like the way I2S interrupt is handled below
> 
> I don't follow this comment. Are you asking to use the macro for defining the IRQ for other blocks as well -- the ones that you previously hard coded?
> 
Yes. That is what i meant.

> All of the IRQ definitions should go away from here and be moved to DTS, like what @galak has been doing, for example, in #12847 except that it should be done for all 3 values and them assembled using the macro.
> It's for another PR.
Yes. We wanted this to be done in DTS too but that involved changing the scripts which generates the macros and we were not so sure about it. Moreover, it was only s1000 which was using a multilevel interrupt structure at that time. It may now be worthwhile to do that as there are a couple more SoCs who are using the multilevel interrupt feature.

Reply:
I already split the macro definition and the usage for I2S.
Maybe you can take care of the move to DTS.
See https://github.com/zephyrproject-rtos/zephyr/pull/12847#issuecomment-459482382